### PR TITLE
Fix type annotation for inner scoverage module

### PR DESCRIPTION
**Pull request by lefou**
* [x] Fix https://github.com/com-lihaoyi/mill/issues/2579

I also changed from `lazy val` to `def` to make overridding and `super.scoverageData`-calls easier.

### CODE REVIEW

**Review by lihaoyi — APPROVED** [review]

**Comment by lefou** [comment]
I added a reproduction for issues
* https://github.com/com-lihaoyi/mill/issues/2579
* https://github.com/com-lihaoyi/mill/issues/2582

File: integration/feature/scoverage/repo/build.sc [hunk 1]
```diff
@@ -0,0 +1,174 @@
+// Reproduction of issue https://github.com/com-lihaoyi/mill/issues/2579
```
Comment (lihaoyi) [inline integration/feature/scoverage/repo/build.sc:1]: Can we minimize this? The external plugins will cause headaches in future if we let them get scattered throughout the codebase, but even ignoring those this test suite  is way larger than what I would expect is needed to reproduce the two issues.

Comment (lefou) [reply integration/feature/scoverage/repo/build.sc:1]: Sure, this is WIP, to find the issue soon.

File: integration/feature/scoverage/repo/build.sc [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// mill plugins
+import $ivy.`com.lihaoyi::mill-contrib-scoverage:`
+import $ivy.`com.github.lolgab::mill-mima::0.0.23`
+
+// imports
+import mill._
+import mill.contrib.scoverage.ScoverageModule
+import mill.scalalib._
+
+val baseDir = build.millSourcePath
```
Comment (lefou) [inline integration/feature/scoverage/repo/build.sc:13]: @lihaoyi Once, I remove this line, the test succeeds.

**Comment by lefou** [comment]
I'm going to split this PR up to only fix the scoverage issue. I'll open a new PR for the reproducer of the second issue.

File: integration/feature/scoverage/repo/build.sc (hunk 1 above)
Comment (lefou) [reply integration/feature/scoverage/repo/build.sc:1]: I wonder whether we should add another section "repro" to explicitly keep reproducers for known issues.

**Pull request merged**